### ISSUE TITLE
SpecAugment implementation

### ISSUE THREAD

**Issue by tayciryahmed** [SeanNaren/deepspeech.pytorch#409]
I suggest implementing the idea in this paper: [SpecAugment: A Simple Data Augmentation Method for Automatic Speech Recognition](https://arxiv.org/abs/1904.08779). 

I would like to work on it.

**Comment by buriy** [comment]
Feel free to work on it.
We tried, results without warp aug are controversial so far (not more help than with other augs).
Please also note that on WSJ their results are much worse than on LibriSpeech.

**Comment by Kailegh** [comment]
I was planning to add that Augmentation too, as soon as I get a little more used to this library and run a few test I will try to implement that paper.

**Comment by jybaek** [comment]
We trying too. We have implemented SpecAugment papers in the repository below, and I'm testing various things in this repository. I'll share the results soon. 

https://github.com/shelling203/SpecAugment

**Comment by Kailegh** [comment]
But that implementation is done with tensorflow, and I am not very sure if it would work with the Dataset method since it works with numpy arrays

**Comment by snakers4** [comment]
> We tried, results without warp aug are controversial so far (not more help than with other augs).

To add a grain of salt to this, I actually ran some of these tests in the following environment:
- Complicated dataset (books, phone calls, youtube, etc);
- Even more complicated validation set (out of domain, out of dataset, etc);
- Clean + noisy annotation;

Here is TLDR of my findings:
- Augs actually helped, they reduced CER by 5-7pp on validation (by increasing CER by 10-15pp on train);
- The problem is that in our case train and val are really different;
- We are talking about CERs around 40-50% on validation;
- We also used a hack to approximate 8kHz files - we dropped 50% of higher frequencies from stft;
- I believe this hack helped in our case much more than any of the augs;
- Also note - I also used dropout, did not use warping (only block dropout);
- And also note - the model datasets used in papers - are VERY clean compared to real life data;

If you need the files
- [Dataset](https://github.com/snakers4/deepspeech.pytorch/blob/master/data/data_loader_aug.py);
- Audio [augs](https://github.com/snakers4/deepspeech.pytorch/blob/master/data/audio_aug.py), spectrogram [augs](https://github.com/snakers4/deepspeech.pytorch/blob/master/data/spectrogram_aug.py);
- STFT [hack](https://github.com/snakers4/deepspeech.pytorch/blob/master/data/data_loader_aug.py#L261);
